tools/clang/include/clang/SPIRV/SpirvInstruction.h

```diff
@@ -1889,6 +1890,40 @@ class SpirvDebugCompilationUnit : public SpirvDebugInstruction {
   spv::SourceLanguage lang;
 };
 
+class SpirvDebugFunctionDeclaration : public SpirvDebugInstruction {
```

Comment:
In general, as I mentioned in an earlier review:
the way the SPIR-V backend visits functions is by starting at the entry-point function and then visits function definitions for any function that is in the call tree from the entrypoint. So I think we will never visit a function declaration on its own, instead we always visit the function definition. Therefore I **think** we'll never emit the `DebugFunctionDeclration` instruction. If that's the case, we could remove this class. It's ok to keep it for now.